### ISSUE TITLE
Too many arguments for `type[tuple[Any, ...]]`

### ISSUE THREAD

**Issue by metaist** [astral-sh/ty#1649]
### Summary

https://play.ty.dev/76cd71f0-4755-4309-8277-2ee2740494d6

`tuple[Any, ...]` works fine, but when it's `type[tuple[Any, ...]]` the variadic tuple signature is rejected.

```python
from typing import Any
ok: tuple[Any, ...]
fails: type[tuple[Any, ...]] # too-many-positional-arguments
```
```
error[too-many-positional-arguments]: Too many positional arguments to class `tuple`: expected 1, got 2
 --> test_variadic_tuple_type.py:4:13
  |
3 | ok: tuple[Any, ...]
4 | fails: type[tuple[str, ...]]
  |             ^^^^^^^^^^^^^^^
  |
info: rule `too-many-positional-arguments` is enabled by default
```

### Version

ty 0.0.1-alpha.28

**Comment by carljm** [comment]
Nice timing, I literally just ran into this bug a minute ago myself :) Thanks for the report!
